pkg/controller/elasticsearch/services/services.go

```diff
@@ -148,10 +132,21 @@ func ElasticsearchURL(es esv1.Elasticsearch, pods []corev1.Pod) string {
 	}
 	if schemeChange {
 		// switch to sending requests directly to a random pod instead of going through the service
-		randomPod := pods[rand.Intn(len(pods))] //nolint:gosec
-		if podURL := ElasticsearchPodURL(randomPod); podURL != "" {
-			return podURL
-		}
+		return AttemptRandomElasticsearchPodURL(es, pods)
+	}
+	return ExternalServiceURL(es)
+}
+
+// AttemptRandomElasticsearchPodURL will return a URL to communicate with a random Elasticsearch pod from the
+// given set of pods.  If for some reason a pod URL cannot be generated, the external Elasticsearch
+// service URL will be returned.
+func AttemptRandomElasticsearchPodURL(es esv1.Elasticsearch, pods []corev1.Pod) string {
```

Comment:
I know naming is hard: how about simply `ElasticsearchURL` and you are already describing in the Godoc that it can fall back to the service URL.

Reply:
https://github.com/elastic/cloud-on-k8s/blob/master/pkg/controller/elasticsearch/services/services.go#L140

Thanks for the suggestion, unfortunately we already have `ElasticsearchURL`, and it conditionally uses this func in this branch.

Reply:
Updated per discussion.